src/fastpm.c

```diff
@@ -370,10 +370,12 @@ int run_fastpm(FastPMConfig * config, RunData * prr, MPI_Comm comm) {
     LEAVE(evolve);
 
     free(time_step);
-    if(usmesh)
-        fastpm_usmesh_destroy(usmesh);
-
-    free(usmesh);
+    for(int si = FASTPM_SOLVER_NSPECIES - 1; si >= 0; si--) {
+        if(usmesh[si]){
```

Comment:
Oh. I see. I guess using a pointer array allows you to check if a species exists like this.